mlir/lib/Conversion/MemRefToEmitC/MemRefToEmitCPass.cpp

```diff
@@ -30,6 +30,16 @@ struct ConvertMemRefToEmitCPass
     : public impl::ConvertMemRefToEmitCBase<ConvertMemRefToEmitCPass> {
   void runOnOperation() override {
     TypeConverter converter;
+    mlir::ModuleOp module = getOperation();
+    module.walk([&](mlir::Operation *op) {
+      if (llvm::isa<mlir::memref::AllocOp, mlir::memref::CopyOp>(op)) {
+        OpBuilder builder(module.getBody(), module.getBody()->begin());
+        builder.create<emitc::IncludeOp>(module.getLoc(),
+                                         builder.getStringAttr("stdlib.h"));
```

Comment:
If one wants to generate C++ code this would need to be
```suggestion
                                         builder.getStringAttr("cstdlib"));
```
Maybe it is worth to add an optional flag to determine if the pass should lower to C or C++.

Furthermore, this should probably have the `is_standard_include` attribute set (https://mlir.llvm.org/docs/Dialects/EmitC/#attributes-14).

is_standard_include

Reply:
IIUC C-style `#include "stdlib.h"` works for C++, only difference is that functions are placed in the global namespace rather `std`.
I'm +1 on adding a flag, not sure it should be part of this commit as it should enforce unrelated C++ restrictions, e.g. multiple return values and tensor types. Can we leave a TODO here for when we have such a flag?